using a bare target name instead of a generator allows cmake to wrap it in CROSSCOMPILING_EMULATOR

--- a/source/blender/makesdna/intern/CMakeLists.txt
+++ b/source/blender/makesdna/intern/CMakeLists.txt
@@ -105,8 +105,7 @@
     ${CMAKE_CURRENT_BINARY_DIR}/dna_type_offsets.h
     ${CMAKE_CURRENT_BINARY_DIR}/dna_verify.c
   COMMAND
-    ${CMAKE_COMMAND} -E env ${PLATFORM_ENV_BUILD}
-    "$<TARGET_FILE:makesdna>"
+    makesdna
     ${CMAKE_CURRENT_BINARY_DIR}/dna.c
     ${CMAKE_CURRENT_BINARY_DIR}/dna_type_offsets.h
     ${CMAKE_CURRENT_BINARY_DIR}/dna_verify.c
--- a/source/blender/makesrna/intern/CMakeLists.txt
+++ b/source/blender/makesrna/intern/CMakeLists.txt
@@ -459,8 +459,7 @@
 add_custom_command(
   OUTPUT ${GENSRC}
   COMMAND
-    ${CMAKE_COMMAND} -E env ${PLATFORM_ENV_BUILD}
-    "$<TARGET_FILE:makesrna>" ${CMAKE_CURRENT_BINARY_DIR}/ ${CMAKE_CURRENT_BINARY_DIR}/../
+    makesrna ${CMAKE_CURRENT_BINARY_DIR}/ ${CMAKE_CURRENT_BINARY_DIR}/../
   DEPENDS makesrna
 )
 
--- a/build_files/cmake/macros.cmake
+++ b/build_files/cmake/macros.cmake
@@ -990,7 +990,7 @@
   add_custom_command(
     OUTPUT ${file_to}
     COMMAND ${CMAKE_COMMAND} -E make_directory ${_file_to_path}
-    COMMAND "$<TARGET_FILE:datatoc>" ${file_from} ${file_to}
+    COMMAND datatoc ${file_from} ${file_to}
     DEPENDS ${file_from} datatoc)
 
   set_source_files_properties(${file_to} PROPERTIES GENERATED TRUE)
@@ -1017,7 +1017,7 @@
   add_custom_command(
     OUTPUT  ${_file_to}
     COMMAND ${CMAKE_COMMAND} -E make_directory ${_file_to_path}
-    COMMAND "$<TARGET_FILE:datatoc>" ${_file_from} ${_file_to}
+    COMMAND datatoc ${_file_from} ${_file_to}
     DEPENDS ${_file_from} datatoc)
 
   set_source_files_properties(${_file_to} PROPERTIES GENERATED TRUE)
@@ -1045,8 +1045,8 @@
   add_custom_command(
     OUTPUT  ${_file_to}
     COMMAND ${CMAKE_COMMAND} -E make_directory ${_file_to_path}
-    COMMAND "$<TARGET_FILE:glsl_preprocess>" ${_file_from} ${_file_tmp}
-    COMMAND "$<TARGET_FILE:datatoc>" ${_file_tmp} ${_file_to}
+    COMMAND glsl_preprocess ${_file_from} ${_file_tmp}
+    COMMAND datatoc ${_file_tmp} ${_file_to}
     DEPENDS ${_file_from} datatoc glsl_preprocess)
 
   set_source_files_properties(${_file_tmp} PROPERTIES GENERATED TRUE)
@@ -1076,8 +1076,7 @@
     COMMAND ${CMAKE_COMMAND} -E
     make_directory ${_file_to_path}
 
-    COMMAND ${CMAKE_COMMAND} -E
-    env ${PLATFORM_ENV_BUILD} "$<TARGET_FILE:msgfmt>" ${_file_from} ${_file_to}
+    COMMAND msgfmt ${_file_from} ${_file_to}
 
     DEPENDS msgfmt ${_file_from})
 
